base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -170,6 +170,28 @@ impl AggregateBody {
             .any(|input| self.outputs.iter().any(|o| o.is_equal_to(input)))
     }
 
+    /// This will check that there are no duplicated inputs. If the body is not sorted, this will sort the body
+    pub fn contains_duplicated_inputs(&mut self) -> bool {
```

Comment:
This can be achieved without the need for `mut`. Later in the PR you convert a borrow into a `mut` which I'm surprised rust lets you do. I think a better approach would be to either do a n^2 search for duplicates or to make a reference array that is sorted `(n*log(n))`.

Alternatively, you could check if it's already sorted and do the linear check, and do the expensive n^2 check if it's not.

Reply:
Isn’t it making a mutable copy? The original remains